src/binary-writer.cc

```diff
@@ -1514,6 +1516,155 @@ Result BinaryWriter::WriteModule() {
       }
     }
     EndSubsection();
+
+    size_t named_types = 0;
```

Comment:
the code blocks below all look very similar.. is it practical to pull this into a function?

Reply:
Ahh yes... I guess I'll have to you one of those new-fangled template things I've heard about.

Reply:
Done!